### DISCUSSION TITLE
db.sqlite3 support

### DISCUSSION THREAD

**Discussion by tradexsrl** [dani-garcia/vaultwarden#4119] [Q&A]
Hi , i have install vaultwarden on my host and i run a docker container from @bruceforce vaultwarden-backup solution but when it runs i get that db.sqlite3 is not found and goes in loop filling the log file
does your docker image contains this utility ?

**Comment by BlackDex** [comment]
The Vaultwarden container does not contain the sqlite3 application.

But that message indicates that it can't find the database. So you probably didn't mounted the correct volume.

**Comment by tradexsrl** [comment]
this is my docker-compose.yml for the container that backup vaultwarden:
version: "3.8"
services:
  vw_backup:
    image: bruceforce/vaultwarden-backup
    container_name: vaultwarden-backup
    restart: unless-stopped
    init: true
    volumes:
      - /volume1/docker/vaultwarden:/data/
      - /volume1/docker/vaultwarden-backup/backups:/backups/
      - /volume1/docker/vaultwarden-backup/logs:/logs/
    environment:
      - BACKUP_DIR=/backups/
      - LOG_DIR=/logs/
      # EVERY DAY @ 5am
      - CRON_TIME=0 5 * * *
      - DELETE_AFTER=30
      - TIMESTAMP=true
      - UID=0 # enter user that has docker permission
      - GID=0 # enter group ID for docker
      - BACKUP_ADD_DATABASE=true
      - BACKUP_ADD_ATTACHMENTS=true
      - BACKUP_ADD_CONFIG_JSON=true
      - BACKUP_ADD_ICON_CACHE=true
      - BACKUP_ADD_RSA_KEY=true
      - LOG_LEVEL=INFO
      #- TZ= <your timezome>

this is a part of docker compose yml from vaultwarden in production:
 volumes:
      - /volume1/docker/vaultwarden/db:/var/lib/postgresql/data:rw
vaultwarden:
    image: vaultwarden/server:latest
    container_name: Vaultwarden
    hostname: vaultwarden
 volumes:
      - /volume1/docker/vaultwarden/data:/data:rw

**Comment by yielddingdong** [comment]
I got the same issue with the same docker-compose (up)

**Comment by BlackDex** [comment]
It looks to me that Postgresql is used as a database, but that backup container expects sqlite. That is probably the issue.

**Reply by yielddingdong** [reply]
![image](https://github.com/user-attachments/assets/30d7de90-a037-476c-b44c-40b394efe570)

But when i see the files, it seems it db.sqlite3 

Is it possible the vaultwarden fils need more right ?

**Reply by BlackDex** [reply]
Not sure what your issue is, but it probably isn't exactly the same as the OP. That looks like PostgreSQL and the Backup container wanting to have a sqlite database.

Your's is probably something different.
I'm not a fan of hijacking posts, but post your logs of vaultwarden here and lets see.

**Reply by yielddingdong** [reply]
there is the backup log : 

2024-10-01 09:03:32 - INFO - Container started
2024-10-01 09:03:32 - INFO - Running vaultwarden-backup version latest@2024-09-01T02:01:19+00:00
2024-10-01 09:03:32 - INFO - Log level set to INFO
2024-10-01 09:03:32 - INFO - Creating first backup on startup.
2024-10-01 09:03:32 - CRITICAL - Database /data/db.sqlite3 not found! Please check if you mounted the vaultwarden volume (in docker-compose or with '--volumes-from=vaultwarden'!)
Exiting

**Reply by BlackDex** [reply]
Well, if Vaultwarden works just fine. And that backup container does not. Then i suggest to ask questions at the maintainer of the backup container.

**Comment by FlavioB79** [comment]
Hi all.
Jumping in here as I want to backup my vaultwarden as well. As mine is running as a Docker Container on my Synology NAS, do I understand it correctly that there's no sqlite3 inside the Docker container itself? If so, is it valid to run sqlite3 backup from my Synology's NAS sqlite3?
TIA
